_source/_posts/2022-02-21-build-deploy-node-app-heroku.md

```diff
@@ -0,0 +1,335 @@
+```
+
+This page displays the user information that was retrieved from Okta when the user logged in.
+
+Replace the code in `views/index.jade` with the following:
+
+```jade
+extends layout
+
+block content
+
+  .container-fluid.py-5.bg-dark.text-white-50
+    h1.display-5.fw-bold  Deploy a Node.JS Application to Heroku
+    p.col-md-8.fs-4
+        | A tutorial on how to deploy a 
+        a(href="https://nodejs.org/en/" target="_blank") node.js 
+        | application to 
+        a(href="https://dashboard.heroku.com/apps"  target="_blank") Heroku
+        | .  Secured by 
+        a(href="https://www.okta.com/"  target="_blank") Okta
+        | .  Written by 
+        a(href="https://profile.fishbowlllc.com/"  target="_blank") Nik Fisher.
+  br
+  p
+    | This repository shows you how to use Okta in a Node.js application and how to deploy the application to Heroku.
+  p Truncated for the purposes of the article.
+```
+
+As mentioned before, this is a static page.  Your next step is deploying the application.  But before you move on to that, run the command `npm run start` and navigate to `http://localhost:3000` to view your application and ensure it is working locally.
+
+{% img
+blog/build-deploy-node-app-heroku/HomePage.png
+alt:"Your Home Page"
+width:"100%" %}
+
+## Deploy to Heroku
+
+Make sure all your code is running and checked into your GitHub repository.  Navigate to your [Heroku Dashboard](https://dashboard.heroku.com/apps).  If this is your first time using Heroku you shouldn't have any apps configured yet.  Click on **New** and then **Create New App**.  Name your app `okta-heroku-webapp-{yourUserName}` or another unique name you feel is appropriate.  Press **Create App**.
+
+On the application page for your newly created application find the *Deployment Method* section on the *Deploy* tab.  Click on **GitHub** and connect your Heroku account to your GitHub account.  
+
+Next, in the *Connect to GitHub* section, find the repository you created for this tutorial and press **Connect**.  
+
+{% img
+blog/build-deploy-node-app-heroku/ConnecttoGitHub.png
+alt:"Connect Heroku to GitHub"
+width:"100%" %}
+
+You can enable automatic deploys under *Automatic deploys*.  You can also configure this to wait for CI to pass before deploying the application.  You can configure the CI from your GitHub account.  This step isn't strictly necessary for this tutorial, but seeing how this fits into the CI pipeline is nice.
+
+{% img
+blog/build-deploy-node-app-heroku/EnableAutomaticDeploys.png
+alt:"Enable Automatic Deploys"
+width:"100%" %}
+
+Finally, under *Manual Deploy* choose the branch you wish to deploy and click **Deploy Branch**.  Wait a moment and you should receive a message from Heroku saying that your app was successfully deployed.
+
+Click the view button to be taken to your application.  At this point, your application won't be running because it isn't configured correctly.  Make note of the URL of your site and makes a few changes.
```

Comment:
```suggestion
Click the view button to be taken to your application.  At this point, your application won't be running because it isn't configured correctly yet.  Make note of the URL of your site and makes a few changes.
```
nuance